source/fundamentals/data-formats/documents.txt

```diff
@@ -278,12 +283,77 @@ The output should resemble the following:
 :java-docs:`BsonDocument API documentation </apidocs/bson/org/bson/BsonDocument.html>`.
 
+JsonObject
+-------------
+
+The ``JsonObject`` class acts as a wrapper for JSON strings.
+If you only want to work with JSON data, you can use ``JsonObject``
+to avoid unnecessary data conversion to a ``Map`` object.
+
+By default, ``JsonObject`` stores :manual:`Extended JSON </reference/mongodb-extended-json/>`.
+You can customize the format of JSON in ``JsonObject`` by specifying a
+:java-docs:`JsonObjectCodec</apidocs/bson/org/bson/codecs/JsonObjectCodec.html>`
+and passing it a
+:java-docs:`JsonWriterSettings</apidocs/bson/org/bson/json/JsonWriterSettings.html>`
+object. For more information on JSON formats, see
+our :doc:`Extended JSON guide </fundamentals/data-formats/document-data-format-extended-json/>`.
+For more information on specifying codecs, see our 
+:doc:`Codecs guide </fundamentals/data-formats/codecs>`.
+
+In the following code snippet, we show how to instantiate a sample ``JsonObject`` 
+instance wrapping an Extended JSON string containing different types of key value pairs:
+
+.. code-block:: java
+
+   String ejsonStr = "{\"_id\": {\"$oid\": \"6035210f35bd203721c3eab8\"},"
+           + "\"name\": \"Gabriel Garc\\u00eda M\\u00e1rquez\","
+           + "\"dateOfDeath\": {\"$date\": \"2014-04-17T04:00:00Z\"},"
+           + "\"novels\": ["
+           + "{\"title\": \"One Hundred Years of Solitude\",\"yearPublished\": 1967},"
+           + "{\"title\": \"Chronicle of a Death Foretold\",\"yearPublished\": 1981},"
+           + "{\"title\": \"Love in the Time of Cholera\",\"yearPublished\": 1985}]}";
+   JsonObject author = new JsonObject(ejsonStr);
+
+To insert this document into a collection, instantiate a collection using
+:java-docs:`getCollection() </apidocs/mongodb-driver-sync/com/mongodb/client/MongoDatabase.html#getCollection(java.lang.String,java.lang.Class)>`,
+specifying the ``JsonObject`` class as the ``documentClass`` parameter. Then, call the
+:doc:`insertOne </usage-examples/insertOne>` operation as follows:
+
+.. code-block:: java
+
+   // MongoClient mongoClient = <code to instantiate your client>;
+
+   MongoDatabase database = mongoClient.getDatabase("fundamentals_data");
+   MongoCollection<JsonObject> collection = database.getCollection("authors", JsonObject.class);
+
+   InsertOneResult result = collection.insertOne(author);
+
+Once you perform a successful insert, you can retrieve the sample JSON data from the 
+collection. While you can use any class that extends ``Bson`` to specify your query,
+here is how to query your data using a ``JsonObject``:
+
+.. code-block:: java
+
+   // MongoClient mongoClient = <code to instantiate your client>;
+
+   JsonObject query = new JsonObject("{\"name\": \"Gabriel Garc\\u00eda M\\u00e1rquez\"}");
+   JsonObject jsonResult = collection.find(query).first();
+   if (jsonResult != null) {
+       System.out.println("extended json string: " + jsonResult.getJson());
```

Comment:
```suggestion
       System.out.println("query result in extended json format: " + jsonResult.getJson());
```

Reply:
If you accept this suggestion, please update the output on line 352 too

Reply:
Thanks for the note. I agree that language is clearer. Updated here: https://github.com/mongodb/docs-java/pull/57/commits/bfc94e306185532004e5159719e3a048ff009a68